### PR TITLE
[SNOW-937596] Add chmod and chown suggestion to config manager message

### PR DESCRIPTION
**Pull request by sfc-gh-lspiegelberg**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   SNOW-937596

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [x] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   When .toml config file has OpenSSL non-compliant permissions (or owners), change warning message to display chown and chmod commands to change file attributes.

### CODE REVIEW

File: src/snowflake/connector/config_manager.py [hunk 1]
```diff
@@ -333,7 +334,14 @@ def read_config(
                     and filep.stat().st_uid != os.getuid()
                 )
             ):
-                warn(f"Bad owner or permissions on {str(filep)}")
+                # for non-Windows, suggest change to 0600 permissions.
+                chmod_message = (
+                    f". To change owner, run chown $USER {str(filep)}. To restrict permissions, run chmod 0600 {str(filep)}."
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/config_manager.py:339]: Surround the commands with some sort of quotation marks, please
```suggestion
                    f". To change owner, run `chown $USER {str(filep)}`. To restrict permissions, run `chmod 0600 {str(filep)}`."
```

File: src/snowflake/connector/config_manager.py [hunk 2]
```diff
@@ -333,7 +334,14 @@ def read_config(
...
+                    if platform.system() != "Windows"
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/config_manager.py:340]: Could you please use `IS_WINDOWS`? Like we do in the tests

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

File: src/snowflake/connector/config_manager.py (hunk 2 above)
Comment (sfc-gh-lspiegelberg) [reply src/snowflake/connector/config_manager.py]: didn't know this existed in the main source code, great! Thought it was a test flag.

**Review by sfc-gh-mkeller — APPROVED** [review]
:ship: :ship:

**Pull request merged**